paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
+        # have not been changed and dummy atoms not added.
+
+        # noinspection PyTypeChecker
+        structure: pmd.Structure = pmd.load_file(coordinate_path, structure=True)
+        structure.coordinates = aligned_structure["!:DM1&!:DM2"].coordinates
+
+        return structure
+
+    @classmethod
+    def prepare_complex_structure(
+        cls,
+        coordinate_path: str,
+        guest_atom_indices: List[int],
+        guest_orientation_mask: str,
+        pull_distance: float,
+        pull_window_index: int,
+        n_pull_windows: int,
+    ) -> pmd.Structure:
+        """Prepares the coordinates of both the host and the guest molecule in a
+        given APR window.
+
+        Parameters
+        ----------
+        coordinate_path
+            The path to the coordinate file which contains the guest molecule
+            bound to the host.
+        guest_atom_indices
+        guest_orientation_mask
+        pull_distance
+        pull_window_index
+        n_pull_windows
         """
-        Read the YAML recipe for the host and guest.
+
+        # Align the host-guest complex so the first guest atom is at (0, 0, 0) and the
+        # second guest atom lies along the positive z-axis.
+        # noinspection PyTypeChecker
+        structure: pmd.Structure = pmd.load_file(coordinate_path, structure=True)
+
+        (
+            guest_orientation_mask_0,
+            guest_orientation_mask_1,
+        ) = guest_orientation_mask.split(" ")
+
+        aligned_structure = align.zalign(
+            structure, guest_orientation_mask_0, guest_orientation_mask_1
+        )
+
+        # n_pull_windows = self.host_yaml["calculation"]["windows"]["pull"]
+
+        # target_initial = self.guest_yaml["restraints"]["guest"][0]["attach"]["target"]
+        # target_final = self.host_yaml["calculation"]["target"]["pull"]
+
+        target_distance = np.linspace(0.0, pull_distance, n_pull_windows)[
+            pull_window_index
+        ]
+        target_difference = target_distance
+
+        for guest_index in guest_atom_indices:
+            aligned_structure.atoms[guest_index].xz += target_difference
+
+        return aligned_structure
+
+    @staticmethod
+    def add_dummy_atoms_to_structure(
+        structure: pmd.Structure,
+        dummy_atom_offsets: List[np.ndarray],
+        offset_coordinates: Optional[np.ndarray] = None,
+    ):
+
+        if offset_coordinates is None:
+            offset_coordinates = np.zeros(3)
+
+        full_coordinates = np.vstack(
+            [
+                structure.coordinates,
+                *[
+                    offset_coordinates + dummy_atom_offset
+                    for dummy_atom_offset in dummy_atom_offsets
+                ],
+            ]
+        )
+
+        for index in range(len(dummy_atom_offsets)):
+            structure.add_atom(pmd.Atom(name="DUM"), f"DM{index + 1}", 1)
+
+        structure.positions = full_coordinates
+
+    # def __init__(self, host, guest=None,
```

Comment:
Safe to delete this entire block?

Reply:
Should be! This will now be handled in `evaluator` when properties are loaded from `taproom`.